storage/map_storage.go

```diff
@@ -20,6 +20,19 @@ import (
 	"github.com/google/trillian"
 )
 
+// ReadOnlyMapTX provides a read-only view into log data.
+// A ReadOnlyMapTX, unlike ReadOnlyMapTreeTX, is not tied to a particular tree.
+type ReadOnlyMapTX interface {
+	DatabaseChecker
```

Comment:
Done.

I've moved DatabaseChecker out of TX and to Storage interfaces. I've elected to keep ReadOnlyMapTX, even though it does nothing currently, as I hope it'll be an incentive to adding methods to it when the time comes. (I really don't want another treeID refactor in my hands.)